src/aktualizr_repo/repo.cc

```diff
@@ -181,21 +181,26 @@ void Repo::addImage(const boost::filesystem::path &image_path) {
 }
 
 void Repo::addTarget(const std::string &target_name, const std::string &hardware_id, const std::string &ecu_serial) {
-  Json::Value image_targets = Utils::parseJSONFile(path_ / "repo/image/targets.json")["signed"];
+  const boost::filesystem::path current = path_ / "repo/director/targets.json";
+  const boost::filesystem::path staging = path_ / "repo/director/staging/targets.json";
+
+  const Json::Value image_targets = Utils::parseJSONFile(path_ / "repo/image/targets.json")["signed"];
   if (!image_targets["targets"].isMember(target_name)) {
     throw std::runtime_error("No such " + target_name + " target in the image repository");
   }
-  Json::Value director_targets = Utils::parseJSONFile(path_ / "repo/director/targets.json")["signed"];
-  if (boost::filesystem::exists(path_ / "repo/director/staging/targets.json")) {
-    director_targets = Utils::parseJSONFile(path_ / "repo/director/staging/targets.json");
+  Json::Value director_targets;
+  if (boost::filesystem::exists(staging)) {
+    director_targets = Utils::parseJSONFile(staging);
+  } else if (!boost::filesystem::exists(current)) {
```

Comment:
nitpick readability:

```
if (happy1) {
} else if (happy2) {
} else {
// sad
}
```

Instead of sad in the middle.

Reply:
Fair enough. Done.